### PR TITLE
Http error codes

### PR DESCRIPTION
**Pull request by phil303**
This is actually two separate commits. One to fix `postRouteServerMiddleware` not being called. The other is giving a setStatus action and its relevant reducer.

:eyeglasses: @nramadas @schwers

### CODE REVIEW

File: lib/modules/reducer.js [hunk 1]
```diff
@@ -13,20 +13,31 @@ const DEFAULT = {
...
 
-      const pageData = {
+      const {
         url,
         urlParams,
         queryParams,
         hashParams,
-        referrer: referrer || referrerFromHistory,
-      };
+        referrer=referrerFromHistory,
```
Comment (nramadas) [inline lib/modules/reducer.js]: do we not need the referrer from the payload anymore?

Comment (nramadas) [inline lib/modules/reducer.js]: derp, now i get it

File: lib/modules/reducer.js [hunk 2]
```diff
@@ -13,20 +13,31 @@ const DEFAULT = {
 
 export default (state=DEFAULT, action={}) => {
   switch (action.type) {
+    case actions.SET_STATUS: {
+      const pageData = merge(state.currentPage, { status: action.status });
+      const before = state.history.slice(0, state.currentPageIndex);
```
Comment (nramadas) [inline lib/modules/reducer.js]: would this be cleaner?

```
const history = [...state.history];
history[state.currentPageIndex] = pageData;
```

Comment (phil303) [inline lib/modules/reducer.js]: yes! good call.

File: lib/modules/reducer.js [hunk 3]
```diff
@@ -35,6 +46,7 @@ export default (state=DEFAULT, action={}) => {
         currentPage: pageData,
       };
     }
+
```
Comment (nramadas) [inline lib/modules/reducer.js:48]: 🔕  do we have any rules in the styleguide regarding newlines between case statements?

Comment (phil303) [inline lib/modules/reducer.js:48]: I'll look into it. As a personal preference, I like to give these case statements a little breathing room. It helps to mentally separate them in the case their contents have newline spaces as well.

**Comment by nramadas** [comment]
LGTM. :fish: 🍣

File: lib/modules/reducer.js (hunk 3 above)
Comment (phil303) [inline lib/modules/reducer.js:48]: following up - not seeing anything in our style guide about this.

**Pull request merged**